### PR TITLE
fix (iOS): Only try to make the alert window key if the app recognizes it

### PR DESCRIPTION
**Pull request by amgleitman**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [x] I am making a change required for Microsoft usage of react-native

## Summary

This is a follow-up to #917.

In iOS 14, if we call `-[UIWindow makeKeyAndVisible]` from a `UIWindow` that isn't part of the app, the current key window gets made non-key. This can result in `RCTKeyWindow()`, and thusly, `RCTPresentedViewController()`, to come back as `nil`, which means that the alert never shows up.

The fix here is to ensure that we don't try to make the alert window key unless it's actually tracked by the application.

## Test Plan

Validated in a downstream app on iOS 14.

### CODE REVIEW

File: React/CoreModules/RCTAlertController.m [hunk 1]
```diff
@@ -32,12 +32,16 @@ - (UIWindow *)alertWindow
 
 - (void)show:(BOOL)animated completion:(void (^)(void))completion
 {
-  [self.alertWindow makeKeyAndVisible];
+  // Call self.alertWindow to ensure that it gets populated
+  UIWindow *alertWindow = self.alertWindow;
 
   // [TODO(macOS GH#774)
   // If the window is tracked by our application then it will show the alert
```
Comment (Saadnajmi) [inline React/CoreModules/RCTAlertController.m:39]: Encapsulate within the TODO macOS tag?

File: React/CoreModules/RCTAlertController.m [hunk 2]
```diff
@@ -32,12 +32,16 @@ - (UIWindow *)alertWindow
...
-  if ([[[UIApplication sharedApplication] windows] containsObject:self.alertWindow]) {
-    [self.alertWindow.rootViewController presentViewController:self animated:animated completion:completion];
+  if ([[[UIApplication sharedApplication] windows] containsObject:alertWindow]) {
+    // On iOS 14, makeKeyAndVisible should only be called if alertWindow is tracked by the application.
+    // Later versions of iOS appear to already do this check for us behind the scenes.
+    [alertWindow makeKeyAndVisible];
+    [alertWindow.rootViewController presentViewController:self animated:animated completion:completion];
```
Comment (Saadnajmi) [inline React/CoreModules/RCTAlertController.m:44]: Sounds like we could put an iOS 14 availability check here to make it easier to remove the extra line once we drop support?

File: React/CoreModules/RCTAlertController.m (hunk 1 above)
Comment (amgleitman) [reply React/CoreModules/RCTAlertController.m:39]: Yeah, seems like a good idea.

File: React/CoreModules/RCTAlertController.m (hunk 2 above)
Comment (amgleitman) [reply React/CoreModules/RCTAlertController.m:44]: I mean, we still want to call `[alertWindow makeKeyAndVisible]`, and it seems like it doesn't hurt for us to do this check anyway.

**Review by Saadnajmi — APPROVED** [review]

**Review by HeyImChris — APPROVED** [review]

**Pull request merged**